File: natour-backend/src/main/java/com/natour/natour/model/dto/ChatResponseDto.java
package com.natour.natour.model.dto;

import java.util.List;

import lombok.Data;

@Data
public class ChatResponseDto {
    private long id;
    private List<ChatMessageDto> messages;
    private Integer totalUnreadMessages;
    private String usernameUser1;
    private String usernameUser2;
    private Long idTrail;

    public ChatResponseDto(
        Long id, 
        List<ChatMessageDto> messages,
        String usernameUser1,
        String usernameUser2,
        Long idTrail
    ) {
        this.id = id;
        this.messages = messages;
        this.usernameUser1 = usernameUser1;
        this.usernameUser2 = usernameUser2;
        this.idTrail = idTrail;
    }
}

File: natour-android/app/src/main/java/com/example/natour/data/dto/ChatResponseDto.kt
package com.example.natour.data.dto

import com.example.natour.data.model.ChatMessage

data class ChatResponseDto(
    val id: Long,
    val idTrail: Long,
    val messages: List<ChatMessage>,
    val totalUnreadMessages: Int,
    val usernameUser1: String,
    val usernameUser2: String
)
